Assets/MixedRealityToolkit/_Core/Devices/BaseController.cs

```diff
@@ -141,5 +152,50 @@ public void AssignControllerMappings(MixedRealityInteractionMapping[] mappings)
         {
             Interactions = mappings;
         }
+
+        private void TryRenderControllerModel(Type controllerType)
+        {
+            GameObject controllerModel = null;
+
+            // Get the global controller model for each hand.
+            if (ControllerHandedness == Handedness.Left && MixedRealityManager.Instance.ActiveProfile.ControllerMappingProfile.GlobalLeftHandModel != null)
+            {
+                controllerModel = MixedRealityManager.Instance.ActiveProfile.ControllerMappingProfile.GlobalLeftHandModel;
+            }
+            else if (ControllerHandedness == Handedness.Right && MixedRealityManager.Instance.ActiveProfile.ControllerMappingProfile.GlobalRightHandModel != null)
+            {
+                controllerModel = MixedRealityManager.Instance.ActiveProfile.ControllerMappingProfile.GlobalRightHandModel;
+            }
+
+            // If a specific controller template wants to override the global model, assign that instead.
+            if (MixedRealityManager.Instance.ActiveProfile.IsControllerMappingEnabled &&
```

Comment:
Move this section above the previous, we should check for an override first and THEN use the global if none found

Reply:
Will do.